sentinelsat/sentinel.py

```diff
@@ -1165,22 +1165,29 @@ def check_files(self, paths=None, ids=None, directory=None, delete=False):
 
         return corrupt
 
-    def _md5_compare(self, file_path, checksum, block_size=2 ** 13):
+    def _checksum_compare(self, file_path, product_info, block_size=2 ** 13):
         """Compare a given MD5 checksum with one calculated from a file."""
+        if "md5" in product_info:
```

Comment:
Maybe reverse the order of the branches to prefer `sha3-256` in case both are present?